stepcvt/project.py

```diff
@@ -30,9 +30,30 @@ def __init__(self, name: str = "", path: Path = None, partinfo: list = None):
         self.path = path
         self.partinfo = [] if partinfo is None else partinfo
 
-    def to_dict(self):
-        return {"type": "CADSource"}
-
+    def to_dict(self, root=None):     
+        path = PurePath(self.path)
+        drive = path.drive
+
+        if path.is_absolute():
+            if not root:
+                raise AssertionError("Root must be provided for absolute paths!")
+            if drive:
+                path = path.relative_to(drive / root)
+
+        return {"type": "CADSource",
+                "name": self.name,
+                "path": str(path.as_posix()),
+                "partinfo": self.partinfo
+                }
+    
+    def from_dict(self, root=None):       
+        if root:
+            path = Path("C:" / root / self["path"])
```

Comment:
I think you can remove the drive here, since it'll be present in root if needed.